### ISSUE TITLE
pip installed podpac (v0.3.0) missing nsidc_smap_opendap_url.txt

### ISSUE THREAD

**Issue by trey-stafford** [creare-com/podpac#247]
**Description**
Version 0.3.0 of podpac installed via pip does not include datalib/nsidc_smap_opendap_url.txt

**Steps to Reproduce**
```
$ pip install podpac[all]
$ pip show podpac
Name: podpac
Version: 0.3.0
Summary: Pipeline for Observational Data Processing, Analysis, and Collaboration
Home-page: https://github.com/creare-com/podpac
Author: Creare
Author-email: UNKNOWN
License: APACHE 2.0
Location: /home/trst2284/miniconda3/envs/podpac/lib/python3.7/site-packages
Requires: scipy, matplotlib, requests, xarray, lazy-import, pint, traitlets, numpy
Required-by:
```

**Expected Behavior**
I expect to be able to use `podpac.datalib.SMAP` to retrieve data from NSIDC's OpenDAP instance.

**Observed Behavior**
Observed in https://github.com/creare-com/podpac-examples/blob/develop/notebooks/basic_examples/010-retrieving-SMAP-data.ipynb

when running cell 5:

```
WARNING:podpac.datalib.smap:Could not retrieve SMAP url from /home/trst2284/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/datalib/nsidc_smap_opendap_url.txt: [Errno 2] No such file or directory: '/home/trst2284/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/datalib/nsidc_smap_opendap_url.txt'

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/miniconda3/envs/podpac/lib/python3.7/site-packages/traitlets/traitlets.py in get(self, obj, cls)
    527         try:
--> 528             value = obj._trait_values[self.name]
    529         except KeyError:

KeyError: 'base_url'

During handling of the above exception, another exception occurred:

UnboundLocalError                         Traceback (most recent call last)
~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/datalib/smap.py in SMAP_BASE_URL()
    215         if 'https://' in r and 'nsidc.org' in r:
--> 216             if rf != r:
    217                 _logger.warning("Updating SMAP url from PODPAC S3 Server.")

UnboundLocalError: local variable 'rf' referenced before assignment

During handling of the above exception, another exception occurred:

NameError                                 Traceback (most recent call last)
<ipython-input-5-799e196f1ad2> in <module>
      9 
     10 # Evaluate the node to retrieve and interpolate the data
---> 11 smap_data = sm.eval(coords)

~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/core/node.py in wrapper(self, coordinates, output)
    554         key = cache_key
    555         cache_coordinates = coordinates.transpose(*sorted(coordinates.dims)) # order agnostic caching
--> 556         if self.has_cache(key, cache_coordinates) and not self.cache_update:
    557             data = self.get_cache(key, cache_coordinates)
    558             if output is not None:

~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/core/node.py in has_cache(self, key, coordinates)
    499         if self.cache_ctrl is None:
    500             return False
--> 501         return self.cache_ctrl.has(self, key, coordinates=coordinates)
    502 
    503     def rem_cache(self, key, coordinates=None, mode=None, all_cache=False):

~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/core/cache/cache.py in has(self, node, key, coordinates, mode)
    221         for c in self._cache_stores:
    222             if c.cache_modes_matches(set([mode])):
--> 223                 if c.has(node=node, key=key, coordinates=coordinates):
    224                     return True
    225         return False

~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/core/cache/cache.py in has(self, node, key, coordinates)
    679              True if there as a cached object for this node for the given key and coordinates.
    680         '''
--> 681         listing = CacheListing(node=node, key=key, coordinates=coordinates)
    682         paths = glob(self.cache_glob(node, key, coordinates))
    683         for p in paths:

~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/core/cache/cache.py in __init__(self, node, key, coordinates, data)
    322 
    323     def __init__(self, node, key, coordinates, data=None):
--> 324         self._node_def = node.definition
    325         self.key = key
    326         self.coordinate_def = None if coordinates is None else coordinates.json

~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/core/node.py in definition(self)
    373             return ref
    374 
--> 375         add_node(self)
    376 
    377         d = OrderedDict()

~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/core/node.py in add_node(node)
    341 
    342             # get base definition and then replace nodes with references, adding nodes depth first
--> 343             d = node.base_definition
    344             if 'lookup_source' in d:
    345                 d['lookup_source'] = add_node(d['lookup_source'])

~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/datalib/smap.py in base_definition(self)
    979         """ Definition for SMAP node. Sources not required as these are computed.
    980         """
--> 981         d = super(podpac.compositor.Compositor, self).base_definition
    982         d['interpolation'] = self.interpolation
    983         return d

~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/core/node.py in base_definition(self)
    297                 continue
    298 
--> 299             attr = getattr(self, key)
    300 
    301             if isinstance(attr, Node):

~/miniconda3/envs/podpac/lib/python3.7/site-packages/traitlets/traitlets.py in __get__(self, obj, cls)
    554             return self
    555         else:
--> 556             return self.get(obj, cls)
    557 
    558     def set(self, obj, value):

~/miniconda3/envs/podpac/lib/python3.7/site-packages/traitlets/traitlets.py in get(self, obj, cls)
    533                 raise TraitError("No default value found for %s trait of %r"
    534                                  % (self.name, obj))
--> 535             value = self._validate(obj, dynamic_default())
    536             obj._trait_values[self.name] = value
    537             return value

~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/datalib/smap.py in _base_url_default(self)
    763     @tl.default('base_url')
    764     def _base_url_default(self):
--> 765         return SMAP_BASE_URL()
    766     product = tl.Enum(SMAP_PRODUCT_MAP.coords['product'].data.tolist(),
    767                       default_value='SPL4SMAU').tag(attr=True)

~/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/datalib/smap.py in SMAP_BASE_URL()
    223                     _logger.warning("Could not overwrite SMAP url update on disk:" + str(e))
    224     except Exception as e:
--> 225         logger.warning("Could not retrieve SMAP url from PODPAC S3 Server. Using default." + str(e))
    226     _SMAP_BASE_URL = BASE_URL
    227     return BASE_URL

NameError: name 'logger' is not defined
```

**Additional Notes**
If applicable, additional code snippets, screenshots, potential solutions, etc.

**Comment by jmilloy** [comment]
I'm not well-versed with the SMAP datalib, but it appears that part of this is a typo in an otherwise expected error. I've pushed a fix to branch hotfix/smap-logger-225

**Comment by trey-stafford** [comment]
Note that I temporarily fixed this issue for myself by creating the missing file manually.

```
$ touch /home/trst2284/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/datalib/nsidc_smap_opendap_url.txt
$ echo "https://n5eil02u.ecs.nsidc.org/opendap/SMAP" > /home/trst2284/miniconda3/envs/podpac/lib/python3.7/site-packages/podpac/datalib/nsidc_smap_opendap_url.txt
```

**Comment by jmilloy** [comment]
An additional hotfix 73b078e in hotfix/smap-url to handle missing file. It appears that the file should be created by podpac with a default value if it is not present, but it was failing to do so.

**Comment by jmilloy** [comment]
Thanks @trey-stafford, I've taken a quick look and passed this along to @mpu-creare for review.

**Comment by mpu-creare** [comment]
Reviewed and merged into develop. Thanks for the work-around, we'll have this update in the 1.0 release which is around the corner.
